bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/ViewsPreferencePage.java

```diff
@@ -114,6 +116,9 @@ public class ViewsPreferencePage extends PreferencePage implements IWorkbenchPre
 	private boolean highContrastMode;
 
 	private Button themingEnabled;
+	private Button updateOnRuntime;
+	private boolean initialStateUpdateOnRuntime = PrefUtil.getAPIPreferenceStore()
+			.getBoolean(IWorkbenchPreferenceConstants.UPDATE_ON_RUNTIME_KEY);
```

Comment:
Why is this solved different then the themeing property does it? That one is simply retrieved from the preference store when identifying whether the value has changed instead of storing it in a field. In my opinion, that solution is better as this field just serves as an unnecessary and potentially error-prone cache.